pkg/log/logger.go

```diff
@@ -30,13 +33,99 @@ var once sync.Once
 var logger *zap.Logger
 var zapConfigs *zap.Config
 
+// LoggerHandle is used to efficiently look up logger references
+type LoggerHandle struct {
+	name string
+}
+
+func (h LoggerHandle) String() string {
+	return h.name
+}
+
+// Logger names
+const (
+	defaultLog  = "log.level"
+	logPrefix   = "log."
+	levelSuffix = ".level"
+)
+
+var K8Shim = LoggerHandle{name: "k8shim"}
+var Kubernetes = LoggerHandle{name: "kubernetes"}
+var Admission = LoggerHandle{name: "admission"}
+var Test = LoggerHandle{name: "test"}
```

Comment:
If we manage all loggers, we can get faster logger retrieval by defining an extra "int" field in `LoggerHandle` which tells the index of the `*zap.Logger` instance inside a slice. We can still use the map for less commonly used loggers.

Reply:
This doesn't work because the core and shim loggers are managed separately. The performance impact of these changes is already negligible.

Reply:
I tried this locally and it does indeed seem to be faster. Overhead is now in the ballpark of a few ns per log call. The one downside is that there's now some more maintenance required for logging to work properly. I've added a unit test to verify that all the loggers exist and are configured as expected.